### PR TITLE
Fmt apps

### PR DESCRIPTION
**Pull request by richsalz**
run util/openssl-format-source (with -c flag) over the apps.  Fix up a few things by hand.

### CODE REVIEW

**Comment by richsalz** [comment]
ping @ekasper

File: apps/apps.c [hunk 1]
```diff
@@ -265,7 +265,6 @@ int password_callback(char *buf, int bufsiz, int verify, PW_CB_DATA *cb_tmp)
         memcpy(buf, password, res);
         return res;
     }
-
```
Comment (levitte) [inline apps/apps.c]: I wonder what it is that makes certain blank lines "optional"...

File: apps/dsa.c [hunk 2]
```diff
@@ -221,14 +222,14 @@ int dsa_main(int argc, char **argv)
...
+                           "PVK form impossible with public key input\n");
                 EVP_PKEY_free(pk);
                 goto end;
             }
             assert(private);
             i = i2b_PVK_bio(out, pk, pvk_encr, 0, passout);
-        }
-        else if (pubin || pubout)
+        } else if (pubin || pubout)
             i = i2b_PublicKey_bio(out, pk);
```
Comment (levitte) [inline apps/dsa.c]: If it were me, I'd wrap this one in { and } as well, just for consistency with the rest of the `if` structure

File: apps/ecparam.c [hunk 3]
```diff
@@ -349,47 +349,56 @@ int ecparam_main(int argc, char **argv)
         print_bignum_var(out, ec_order, "ec_order", len, buffer);
         print_bignum_var(out, ec_cofactor, "ec_cofactor", len, buffer);
         BIO_printf(out, "    int ok = 0;\n"
```
Comment (levitte) [inline apps/ecparam.c]: I'd like the first string to end up as its own line as well, so it's aligned with the rest of them

File: apps/rehash.c [hunk 4]
```diff
@@ -198,7 +194,7 @@ static int handle_symlink(const char *filename, const char *fullpath)
  */
 static int do_file(const char *filename, const char *fullpath, enum Hash h)
 {
-    STACK_OF (X509_INFO) *inf = NULL;
+    STACK_OF (X509_INFO) * inf = NULL;
```
Comment (levitte) [inline apps/rehash.c]: This should be `STACK_OF(X509_INFO) *inf = NULL`

The reason `util/openssl-format-source` didn't fix this is the space between `STACK_OF` and `(X509_INFO)`...  it shouldn't have been there.

File: apps/req.c [hunk 5]
```diff
@@ -1094,10 +1088,10 @@ static int auto_info(X509_REQ *req, STACK_OF(CONF_VALUE) *dn_sk,
          */
         for (p = v->name; *p; p++) {
 #ifndef CHARSET_EBCDIC
-            spec_char = ((*p == ':') || (*p == ',') || (*p == '.'));
+            spec_char = *p == ':' || *p == ',' || *p == '.';
```
Comment (levitte) [inline apps/req.c]: I would keep the outer parentheses here.

File: apps/req.c [hunk 6]
```diff
@@ -1094,10 +1088,10 @@ static int auto_info(X509_REQ *req, STACK_OF(CONF_VALUE) *dn_sk,
...
 #else
-            spec_char = ((*p == os_toascii[':']) || (*p == os_toascii[','])
-                    || (*p == os_toascii['.']));
+            spec_char = *p == os_toascii[':'] || *p == os_toascii[',']
+                         || *p == os_toascii['.'];
```
Comment (levitte) [inline apps/req.c]: Same here.  WIth all the `=` in there, it's easy to get cross-eyed.  The parentheses help.

File: apps/s_client.c [hunk 7]
```diff
@@ -457,7 +453,7 @@ static ossl_ssize_t checked_uint8(const char **inptr, void *out)
 struct tlsa_field {
     void *var;
     const char *name;
-    ossl_ssize_t (*parser)(const char **, void *);
+     ossl_ssize_t(*parser) (const char **, void *);
```
Comment (levitte) [inline apps/s_client.c]: This is wrong.  You may need to add a line for `ossl_ssize_t` in `util/indent.pro`...

File: apps/s_client.c [hunk 8]
```diff
@@ -1955,15 +1944,15 @@ int s_client_main(int argc, char **argv)
         {
             static const unsigned char tls_do[] = {
                 /* IAC    DO   START_TLS */
-                   255,   253, 46
+                255, 253, 46
```
Comment (levitte) [inline apps/s_client.c]: Please undo.  The numbers are aligned with the words in the comment above.

File: apps/s_client.c [hunk 9]
```diff
@@ -1955,15 +1944,15 @@ int s_client_main(int argc, char **argv)
...
             };
             static const unsigned char tls_will[] = {
                 /* IAC  WILL START_TLS */
-                   255, 251, 46
+                255, 251, 46
```
Comment (levitte) [inline apps/s_client.c]: Please undo.  The numbers are aligned with the words in the comment above.

File: apps/s_client.c [hunk 10]
```diff
@@ -1955,15 +1944,15 @@ int s_client_main(int argc, char **argv)
...
             };
             static const unsigned char tls_follows[] = {
                 /* IAC  SB   START_TLS FOLLOWS IAC  SE */
-                   255, 250, 46,       1,      255, 240
+                255, 250, 46, 1, 255, 240
```
Comment (levitte) [inline apps/s_client.c]: Please undo.  The numbers are aligned with the words in the comment above.

**Comment by levitte** [comment]
Done a read through, left some comments.

File: apps/apps.c (hunk 1 above)
Comment (richsalz) [inline apps/apps.c]: beats me :)

File: apps/dsa.c (hunk 2 above)
Comment (richsalz) [inline apps/dsa.c]: which is what our style says to do :)  fixed.

File: apps/ecparam.c (hunk 3 above)
Comment (richsalz) [inline apps/ecparam.c]: fixed.

File: apps/rehash.c (hunk 4 above)
Comment (richsalz) [inline apps/rehash.c]: Aha.  That’s why.  Fixed.

File: apps/req.c (hunk 5 above)
Comment (richsalz) [inline apps/req.c]: I created a macro, looks simpler IMO.

File: apps/req.c (hunk 6 above)
Comment (richsalz) [inline apps/req.c]: Same macro solution.

File: apps/s_client.c (hunk 7 above)
Comment (richsalz) [inline apps/s_client.c]: Fixed and added the –T flag; thanks!

File: apps/s_client.c (hunk 8 above)
Comment (richsalz) [inline apps/s_client.c]: Oops, fixed.

File: apps/s_client.c (hunk 9 above)
Comment (richsalz) [inline apps/s_client.c]: Restored.

File: apps/s_client.c (hunk 10 above)
Comment (richsalz) [inline apps/s_client.c]: Also restored.

**Comment by richsalz** [comment]
Fixup commit added.  Thanks!

File: apps/dgst.c [hunk 11]
```diff
@@ -39,7 +38,7 @@ typedef enum OPTION_choice {
 OPTIONS dgst_options[] = {
     {OPT_HELP_STR, 1, '-', "Usage: %s [options] [file...]\n"},
     {OPT_HELP_STR, 1, '-',
-        "  file... files to digest (default is stdin)\n"},
+     "  file... files to digest (default is stdin)\n"},
```
Comment (FdaSilvaYY) [inline apps/dgst.c]: Could-it be merged on one line ?

File: apps/cms.c [hunk 12]
```diff
@@ -87,7 +87,7 @@ typedef enum OPTION_choice {
 OPTIONS cms_options[] = {
     {OPT_HELP_STR, 1, '-', "Usage: %s [options] cert.pem...\n"},
     {OPT_HELP_STR, 1, '-',
```
Comment (FdaSilvaYY) [inline apps/cms.c]: Fit on one line ?

File: apps/engine.c [hunk 13]
```diff
@@ -29,7 +29,7 @@ typedef enum OPTION_choice {
 OPTIONS engine_options[] = {
     {OPT_HELP_STR, 1, '-', "Usage: %s [options] engine...\n"},
     {OPT_HELP_STR, 1, '-',
-        "  engine... Engines to load\n"},
+     "  engine... Engines to load\n"},
```
Comment (FdaSilvaYY) [inline apps/engine.c]: Could be merged on one line ?

File: apps/prime.c [hunk 14]
```diff
@@ -20,7 +20,7 @@ typedef enum OPTION_choice {
 OPTIONS prime_options[] = {
     {OPT_HELP_STR, 1, '-', "Usage: %s [options] [number...]\n"},
     {OPT_HELP_STR, 1, '-',
-        "  number Number to check for primality\n"},
+     "  number Number to check for primality\n"},
```
Comment (FdaSilvaYY) [inline apps/prime.c]: Could be merged on one line ?

File: apps/rehash.c [hunk 15]
```diff
@@ -476,4 +473,4 @@ int rehash_main(int argc, char **argv)
     return (1);
 }
 
-#endif /* defined(OPENSSL_SYS_UNIX) || defined(__APPLE__) */
+#endif        /* defined(OPENSSL_SYS_UNIX) || * defined(__APPLE__) */
```
Comment (FdaSilvaYY) [inline apps/rehash.c]: Why adding a star(*) here ?

File: apps/req.c [hunk 16]
```diff
@@ -44,6 +44,17 @@
 #define DEFAULT_KEY_LENGTH      2048
 #define MIN_KEY_LENGTH          512
 
+/* Is this a separator? Always in ASCII */
+#define ASCSEPARATOR(c) (c == ':' || c == ',' || c == '.')
```
Comment (FdaSilvaYY) [inline apps/req.c]: As it is a macro :
+#define ASCSEPARATOR(c) ((c) == ':' || (c) == ',' || (c) == '.')

File: apps/rehash.c (hunk 15 above)
Comment (richsalz) [inline apps/rehash.c]: Because indent broke the line and I carelessly rejoined it.  Fixed.

File: apps/req.c (hunk 16 above)
Comment (richsalz) [inline apps/req.c]: Shrug, why not.  Fixed both macros.

**Comment by richsalz** [comment]
pushed updated commit to do many of the things @FdaSilvaYY found.  As for the "join on one line" things:  I would rather not.  If the tool needs to be fixed, let's try to fix that.  But I'll see what @levitte thinks.

File: apps/testdsa.h [hunk 17]
```diff
@@ -148,7 +147,7 @@ DSA *get_dsa1024()
     q = BN_bin2bn(dsa1024_q, sizeof(dsa1024_q), NULL);
     g = BN_bin2bn(dsa1024_g, sizeof(dsa1024_g), NULL);
     if ((priv_key == NULL) || (pub_key == NULL) || (p == NULL) || (q == NULL)
-            || (g == NULL)) {
+        || (g == NULL)) {
```
Comment (FdaSilvaYY) [inline apps/testdsa.h]: This test is not modified as previous :  ` if (priv_key == NULL `

Comment (richsalz) [inline apps/testdsa.h]: what?

**Comment by mattcaswell** [comment]
What is happening with this?

**Comment by richsalz** [comment]
It never got enough approvals.  It’s less important than other things, so move it post-1.1.0

**Comment by mattcaswell** [comment]
Done

**Pull request closed without merging**